builtin/providers/google/resource_compute_vpn_gateway.go

```diff
@@ -0,0 +1,121 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func resourceComputeVpnGateway() *schema.Resource {
+	return &schema.Resource{
+		// Unfortunately, the VPNGatewayService does not support update
+		// operations. This is why everything is marked forcenew
+		Create: resourceComputeVpnGatewayCreate,
+		Read:	resourceComputeVpnGatewayRead,
+		Delete: resourceComputeVpnGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:	  schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"network": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:	  schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"self_link": &schema.Schema{
+				Type:	  schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceComputeVpnGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	name := d.Get("name").(string)
+	network := d.Get("network").(string)
+	description := d.Get("description").(string)
+	region := d.Get("region").(string)
```

Comment:
yep, I think that's best.

If you'll come up with anything more elegant than calling a function in each of these CRUD functions, it will be even better, but if not, the current solution is :ok_hand: .

Due to the nature of schema (which is typically validated before any API calls are made) it would probably require much bigger changes in the core.